src/Board.cpp

```diff
@@ -0,0 +1,207 @@
+#include "Board.h"
+
+/**
+ * @brief Construct a new Board:: Board object
+ *
+ * @param board_string
+ */
+Board::Board(const string &board_string): m_isWhiteTurn(true) {
+    // Initialize the board
+    for(auto row = 0; row < 8; ++row)
+        for(auto col = 0; col < 8; ++col)
+            m_board[row][col] = nullptr;
+    //set default king positions
+    m_kingPos[0] = {-1, -1};
+    m_kingPos[1] = {-1, -1};
+
+    // Create the pieces
+    ChessPieceFactory factory;
+    for(auto i = 0; i < board_string.size(); ++i) {
+        auto row = i / 8;
+        auto col = i % 8;
+        if(board_string[i] == 'K')
+            m_kingPos[WHITE] = {row, col};
+        else if(board_string[i] == 'k')
+            m_kingPos[BLACK] = {row, col};
+
+        m_board[row][col] = factory.createPiece(board_string[i]);
+    }
+}
+
+/**
+ * @brief get the response code of the move
+ * response code value : 11 - there is not piece at the source
+ *                      12 - the piece in the source is piece of your opponent
+ *                      13 - there one of your pieces at the destination
+ *                      21 - illegal movement of that piece
+ *                      31 - this movement will cause you checkmate
+ *                      41 - the last movement was legal and cause check
+ *                      42 - the last movement was legal
+ *
+ * @param move the move to check
+ * @return int the response code
+ */
+int Board::getResponeCode(const std::string &move) {
```

Comment:
consider breaking down the logic in getResponeCode() into smaller functions to improve readability and maintainability.